src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECCurve.cs

```diff
@@ -142,7 +142,21 @@ public static ECCurve CreateFromValue(string oidValue)
 
         private static ECCurve CreateFromValueAndName(string oidValue, string oidFriendlyName)
         {
-            return ECCurve.Create(new Oid(oidValue, oidFriendlyName));
+            Oid oid = null;
+
+            if (oidValue is null && oidFriendlyName is object)
+            {
+                try
+                {
+                    oid = Oid.FromFriendlyName(oidFriendlyName, OidGroup.All);
```

Comment:
The problem with All is it (used to?) ask Active Directory in certain "not found" fallback paths, making it expensive on failure.

Though... interestingly... if we pass in All with FromFriendlyName, if I'm reading it correctly, we'll ask for `All | DISABLE_SEARCH_DS`, and then fallback is blocked because a) we used FromFriendlyName and b) we said All.

I think that the "don't suppress the AD lookup" path is technically reachable, but one has to try really hard :smile:.

So this may, in fact, just be better.

Reply:
That's not exactly how I am reading it.

1. If the OID group isn't know to suppress AD lookups (`All` will not suppress). Then it will perform the search using `CRYPT_OID_DISABLE_SEARCH_DS_FLAG` first.
2. If that fails, then perform the search without the flag.
3. If the group wasn't `All`, try the search again with the group as `All` if the `fallBackToAllGroups` flag is specified.

So I suspect it will perform an directory services search as a last resort.

If we really wanted to make sure we never hit directory services ever I suppose we could try a try a few different OID groups that are known not to cause a network lookup. It doesn't look like multiple groups can be specified in a single lookup.

Reply:
I actually didn't know that certain groups were known to never cause an AD query until I read this comment a few minutes ago:

https://github.com/dotnet/corefx/blob/124cceaae8309fe0d357e142ae5a0084101107c2/src/Common/src/Interop/Windows/Crypt32/Interop.FindOidInfo.cs#L118

Reply:
For (3) "try again if fallBackToAllGroups is specified", Oid.FromFriendlyName calls it as `string oidValue = OidLookup.ToOid(friendlyName, group, fallBackToAllGroups: false);`

`new Oid(string, string)` uses group All, so it wouldn't fall back (because group was All).  So in order to hit I think one has to

* Oid.FromFriendlyName(someFoundName, OidGroup.Something)
* Change the FriendlyName property

as the only way to get the fallBackToAllGroups: true when group isn't already All.

Reply:
@bartonjs won't this line hit directory services?

https://github.com/dotnet/corefx/blob/124cceaae8309fe0d357e142ae5a0084101107c2/src/Common/src/Interop/Windows/Crypt32/Interop.FindOidInfo.cs#L87

Since that isn't passing in the flag? This will hit `CryptFindOIDInfo` with the `All` group without specifying the flag.

Reply:
Yes, yes it will.  That explains why it doesn't try falling back again for All :)

::whistles innocently::

Reply:
Uh, my pull request really does have a problem! Checkmate!

Wait..